### PR TITLE
Add support for Razer Naga V2 HyperSpeed through Receiver

### LINKED ISSUE #1966: Support for Razer Naga V2 Hyperspeed
### Device Info

- Name: Razer Naga V2 Hyperspeed
- Product Number: RZ01-03600100
- Model Number: RZ01-0360

### Device Information
```Bus 001 Device 007: ID 1532:00b4 Razer USA, Ltd Razer Naga V2 HyperSpeed
Device Descriptor:
  bLength                18
  bDescriptorType         1
  bcdUSB               2.00
  bDeviceClass            0 
  bDeviceSubClass         0 
  bDeviceProtocol         0 
  bMaxPacketSize0        64
  idVendor           0x1532 Razer USA, Ltd
  idProduct          0x00b4 
  bcdDevice            1.00
  iManufacturer           1 Razer
  iProduct                2 Razer Naga V2 HyperSpeed
  iSerial                 3 000000000000
  bNumConfigurations      1
  Configuration Descriptor:
    bLength                 9
    bDescriptorType         2
    wTotalLength       0x0054
    bNumInterfaces          3
    bConfigurationValue     1
    iConfiguration          4 Default configuration
    bmAttributes         0xa0
      (Bus Powered)
      Remote Wakeup
    MaxPower              100mA
    Interface Descriptor:
      bLength                 9
      bDescriptorType         4
      bInterfaceNumber        0
      bAlternateSetting       0
      bNumEndpoints           1
      bInterfaceClass         3 Human Interface Device
      bInterfaceSubClass      1 Boot Interface Subclass
      bInterfaceProtocol      2 Mouse
      iInterface              0 
        HID Device Descriptor:
          bLength                 9
          bDescriptorType        33
          bcdHID               1.11
          bCountryCode            0 Not supported
          bNumDescriptors         1
          bDescriptorType        34 Report
          wDescriptorLength      97
         Report Descriptors: 
           ** UNAVAILABLE **
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x81  EP 1 IN
        bmAttributes            3
          Transfer Type            Interrupt
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x0040  1x 64 bytes
        bInterval               1
    Interface Descriptor:
      bLength                 9
      bDescriptorType         4
      bInterfaceNumber        1
      bAlternateSetting       0
      bNumEndpoints           1
      bInterfaceClass         3 Human Interface Device
      bInterfaceSubClass      1 Boot Interface Subclass
      bInterfaceProtocol      1 Keyboard
      iInterface              0 
        HID Device Descriptor:
          bLength                 9
          bDescriptorType        33
          bcdHID               1.11
          bCountryCode            0 Not supported
          bNumDescriptors         1
          bDescriptorType        34 Report
          wDescriptorLength     314
         Report Descriptors: 
           ** UNAVAILABLE **
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x82  EP 2 IN
        bmAttributes            3
          Transfer Type            Interrupt
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x0040  1x 64 bytes
        bInterval               1
    Interface Descriptor:
      bLength                 9
      bDescriptorType         4
      bInterfaceNumber        2
      bAlternateSetting       0
      bNumEndpoints           1
      bInterfaceClass         3 Human Interface Device
      bInterfaceSubClass      1 Boot Interface Subclass
      bInterfaceProtocol      1 Keyboard
      iInterface              0 
        HID Device Descriptor:
          bLength                 9
          bDescriptorType        33
          bcdHID               1.11
          bCountryCode            0 Not supported
          bNumDescriptors         1
          bDescriptorType        34 Report
          wDescriptorLength      61
         Report Descriptors: 
           ** UNAVAILABLE **
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x83  EP 3 IN
        bmAttributes            3
          Transfer Type            Interrupt
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x0040  1x 64 bytes
        bInterval               1
Device Status:     0x0000
  (Bus Powered)
```

```total 0
drwxr-xr-x 2 root root 300 Nov 29 17:54 .
drwxr-xr-x 4 root root 800 Nov 29 19:13 ..
lrwxrwxrwx 1 root root  10 Nov 29 17:54 usb-046d_Logitech_StreamCam_F0023F65-event-if00 -> ../event13
lrwxrwxrwx 1 root root   9 Nov 29 17:54 usb-04d9_USB-HID_Keyboard-event-if02 -> ../event5
lrwxrwxrwx 1 root root   9 Nov 29 17:54 usb-04d9_USB-HID_Keyboard-event-kbd -> ../event2
lrwxrwxrwx 1 root root   9 Nov 29 17:54 usb-04d9_USB-HID_Keyboard-if02-event-kbd -> ../event6
lrwxrwxrwx 1 root root  10 Nov 29 17:54 usb-Razer_Razer_Naga_V2_HyperSpeed_000000000000-event-if01 -> ../event10
lrwxrwxrwx 1 root root   9 Nov 29 17:54 usb-Razer_Razer_Naga_V2_HyperSpeed_000000000000-event-mouse -> ../event7
lrwxrwxrwx 1 root root   9 Nov 29 17:54 usb-Razer_Razer_Naga_V2_HyperSpeed_000000000000-if01-event-kbd -> ../event8
lrwxrwxrwx 1 root root   9 Nov 29 17:54 usb-Razer_Razer_Naga_V2_HyperSpeed_000000000000-if01-event-mouse -> ../event9
lrwxrwxrwx 1 root root   9 Nov 29 17:54 usb-Razer_Razer_Naga_V2_HyperSpeed_000000000000-if01-mouse -> ../mouse2
lrwxrwxrwx 1 root root  10 Nov 29 17:54 usb-Razer_Razer_Naga_V2_HyperSpeed_000000000000-if02-event-kbd -> ../event11
lrwxrwxrwx 1 root root   9 Nov 29 17:54 usb-Razer_Razer_Naga_V2_HyperSpeed_000000000000-mouse -> ../mouse1
lrwxrwxrwx 1 root root   9 Nov 29 17:54 usb-Valve_Software_Steam_Controller-event-mouse -> ../event3
lrwxrwxrwx 1 root root   9 Nov 29 17:54 usb-Valve_Software_Steam_Controller-mouse -> ../mouse0
```

### PR DESCRIPTION
**Pull request by rhaamo**
First dig into the code, there is probably awful things there...

Will resolves #1966 

I have tried it using the Receiver, using Bluetooth polychromatic doesn't shows a device, I don't really know how to get it further.

For the receiver part:
- battery report is the same as using bluetooth (using a not-full battery to avoid 100%)
- if I change the DPI using the mouse buttons, the right DPI is shown when I relaunch polychromatic
- but changing the DPI in polychromatic or razergenie doesn't do anything
- changing poll rate in razergenie then opening polychromatic shows the right value (and same for other direction) so I guess that's ok ? (idk how to know really)
- I don't really know how to test the rest

I have PCAP and screenshots from synapse available.

```
LANG=C PYTHONPATH="pylib:daemon" polychromatic-cli -k

Razer Naga V2 HyperSpeed (Receiver)
–––––––––––––––––––––––––––––––––––
VID:PID           1532:00B4
Serial            PM2244H26100220
Firmware Version  v1.0
Custom Effects    Unsupported
Battery           90%

Supports DPI (--dpi) between 100 and 30000

Zone (-z)  Option (-o)            Parameters (-p)  Colours (-c)
main       poll_rate              125, 500, 1000
main       idle_time              1 - 15  minutes
main       low_battery_threshold  1 - 100 %
```

### CODE REVIEW

File: daemon/openrazer_daemon/hardware/mouse.py [hunk 1]
```diff
@@ -1860,3 +1860,43 @@ def _close(self):
...
+
+class RazerNagaV2HyperSpeedReceiver(__RazerDevice):
+    """
+    Class for the Razer Naga V2 HyperSpeed (Receiver)
+    """
+    EVENT_FILE_REGEX = re.compile(r'.*usb-Razer_Razer_Naga_V2_HyperSpeed_000000000000-if0(1|2)-event-kbd')
+
+    USB_VID = 0x1532
+    USB_PID = 0x00B4
+    HAS_MATRIX = False
```
Comment (z3ntu) [inline daemon/openrazer_daemon/hardware/mouse.py:1873]: Drop this line if there's no matrix support. But are you sure there isn't? **edit:** seems there's no RGB on this device so then it doesn't have matrix support. Just drop the line.

**Comment by z3ntu** [comment]
Diff looks okay, but please check the CI output to make that happy.

For Bluetooth are there any messages from the driver appearing in the dmesg? Might need to use higher timeouts there.

**Comment by 87dnz** [comment]
i have a the naga v2 hyperspeed, how can i help ?

**Comment by bertrand-benoit** [comment]
i have a the naga v2 hyperspeed, how can i help ?

**Comment by z3ntu** [comment]
@rhaamo Any update here?

**Comment by rhaamo** [comment]
Sorry I don't use that mouse since some time now, but I hadn't any issues with my code in the meantime.
As for the Bluetooth part I have no idea for the support of it.

**Comment by z3ntu** [comment]
For Bluetooth, I'd be fine ripping out Bluetooth support from this PR if it doesn't work.

> but changing the DPI in polychromatic or razergenie doesn't do anything

This would need to be resolved before merging. If you don't have the device anymore (or don't want to work on it anymore), then somebody else needs to pick it up and figure out what's wrong there.

**Comment by w3z315** [comment]
Hey! Any updates on this? May I assist somehow? Feel free to ask. 👍🏻

**Comment by z3ntu** [comment]
Closing in favor of #2141

**Pull request closed without merging**